samples/nrf9160/multicell_location/src/main.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <modem/lte_lc.h>
+#include <dk_buttons_and_leds.h>
+#include <net/multicell_location.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(multicell_location_sample, CONFIG_MULTICELL_LOCATION_SAMPLE_LOG_LEVEL);
+
+static K_SEM_DEFINE(lte_connected, 0, 1);
+static K_SEM_DEFINE(cell_data_ready, 0, 1);
+static struct k_work_delayable periodic_search_work;
+static struct lte_lc_ncell neighbor_cells[17];
+static struct lte_lc_cells_info cell_data = {
+	.neighbor_cells = neighbor_cells,
+};
+
+static void lte_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS:
+		if ((evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_HOME) &&
+		     (evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_ROAMING)) {
+			break;
+		}
+
+		LOG_INF("Network registration status: %s",
+			evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ?
+			"Connected - home network" : "Connected - roaming");
+		k_sem_give(&lte_connected);
+		break;
+	case LTE_LC_EVT_PSM_UPDATE:
+		LOG_INF("PSM parameter update: TAU: %d, Active time: %d",
+			evt->psm_cfg.tau, evt->psm_cfg.active_time);
+		break;
+	case LTE_LC_EVT_EDRX_UPDATE: {
+		char log_buf[60];
+		ssize_t len;
+
+		len = snprintk(log_buf, sizeof(log_buf),
+			       "eDRX parameter update: eDRX: %f, PTW: %f",
+			       evt->edrx_cfg.edrx, evt->edrx_cfg.ptw);
+		if (len > 0) {
+			LOG_INF("%s", log_strdup(log_buf));
+		}
+		break;
+	}
+	case LTE_LC_EVT_RRC_UPDATE:
+		LOG_INF("RRC mode: %s",
+			evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED ?
+			"Connected" : "Idle");
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		LOG_INF("LTE cell changed: Cell ID: %d, Tracking area: %d",
+			evt->cell.id, evt->cell.tac);
+		break;
+	case LTE_LC_EVT_LTE_MODE_UPDATE:
+		LOG_INF("Active LTE mode changed: %s",
+			evt->lte_mode == LTE_LC_LTE_MODE_NONE ? "None" :
+			evt->lte_mode == LTE_LC_LTE_MODE_LTEM ? "LTE-M" :
+			evt->lte_mode == LTE_LC_LTE_MODE_NBIOT ? "NB-IoT" :
+			"Unknown");
+		break;
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS:
+		LOG_INF("Neighbor cell measurements received");
+
+		/* Copy current and neighbor cell information. */
+		memcpy(&cell_data, &evt->cells_info, sizeof(struct lte_lc_cells_info));
+		memcpy(neighbor_cells, evt->cells_info.neighbor_cells,
+			sizeof(struct lte_lc_ncell) * cell_data.ncells_count);
+		cell_data.neighbor_cells = neighbor_cells;
+
+		k_sem_give(&cell_data_ready);
+		break;
+	default:
+		break;
+	}
+}
+
+static int lte_connect(void)
+{
+	if (IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT)) {
+		/* Do nothing, modem is already configured and LTE connected. */
+		return 0;
+	} else {
+		int err;
+
+		err = lte_lc_psm_req(true);
```

Comment:
If `CONFIG_LTE_AUTO_INIT_AND_CONNECT` Is enabled the sample will not request PSM nor eDRX. Also the handler will not be registered that `NCELLMEAS` callbacks depend on. It seems that this sample should probably not support this config. The semaphore that blocks the main thread will also not be given. In the case you decide to remove it the sample should probably have an assert in case its enabled by a user.

Reply:
Added assert to ensure auto connect isn't used in the sample and removed the check for auto connect here